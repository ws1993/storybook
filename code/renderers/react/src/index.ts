/// <reference types="webpack-env" />
import './globals';

export * from './public-types';

export * from './portable-stories';

export * from './preview';

export type { ReactParameters } from './types';

// optimization: stop HMR propagation in webpack

// optimization: stop HMR propagation in webpack
if (typeof module !== 'undefined') {
  module?.hot?.decline();
}
