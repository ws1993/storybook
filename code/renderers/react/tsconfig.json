{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "@storybook/react": ["./src"],
      "@storybook/blocks": ["../../lib/blocks/src"]
    }
  },
  "include": ["src/**/*", "template/**/*"]
}
