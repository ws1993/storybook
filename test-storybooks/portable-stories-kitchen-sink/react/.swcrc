{
  "jsc": {
    "parser": {
      "syntax": "typescript",
      "tsx": true,
      "decorators": true,
      "dynamicImport": true
    },
    "transform": {
      "react": {
        "runtime": "automatic"
      }
    },
    "target": "es2015"
  },
  "module": {
    "type": "es6",
    "noInterop": true
  }
}
