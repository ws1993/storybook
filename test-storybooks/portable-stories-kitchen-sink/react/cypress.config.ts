import { defineConfig } from "cypress";

export default defineConfig({
  screenshotOnRunFailure: false,
  component: {
    devServer: {
      framework: "react",
      bundler: "vite",
    },
  },
});
