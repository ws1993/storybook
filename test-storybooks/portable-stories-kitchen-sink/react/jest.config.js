module.exports = {
  testMatch: ['**/?(*.)+(test).[jt]s?(x)'],
  setupFilesAfterEnv: ['<rootDir>/jest.setup.ts'],
  transform: {
    '^.+\\.(t|j)sx?$': '@swc/jest',
  },
  moduleNameMapper: {
    '\\.css$': 'identity-obj-proxy',
  },
  testEnvironment: 'jsdom',
};
