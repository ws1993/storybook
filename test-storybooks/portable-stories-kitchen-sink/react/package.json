{
  "name": "portable-stories-react",
  "version": "0.0.0",
  "private": true,
  "scripts": {
    "build": "tsc && vite build",
    "cypress": "cypress run --component",
    "cypress-open": "cypress open --component",
    "dev": "vite",
    "jest": "node --experimental-vm-modules node_modules/jest/bin/jest.js",
    "lint": "eslint . --ext ts,tsx --report-unused-disable-directives --max-warnings 0",
    "playwright-ct": "playwright test -c playwright-ct.config.ts",
    "playwright-e2e": "playwright test -c playwright-e2e.config.ts",
    "preview": "vite preview",
    "storybook": "NODE_OPTIONS=\"--preserve-symlinks --preserve-symlinks-main\" storybook dev -p 6006",
    "vitest": "SKIP_FAIL_ON_PURPOSE=true vitest run"
  },
  "resolutions": {
    "@playwright/test": "1.48.1",
    "@storybook/addon-a11y": "file:../../../code/addons/a11y",
    "@storybook/addon-actions": "file:../../../code/addons/actions",
    "@storybook/addon-backgrounds": "file:../../../code/addons/backgrounds",
    "@storybook/addon-controls": "file:../../../code/addons/controls",
    "@storybook/addon-docs": "file:../../../code/addons/docs",
    "@storybook/addon-essentials": "file:../../../code/addons/essentials",
    "@storybook/addon-highlight": "file:../../../code/addons/highlight",
    "@storybook/addon-interactions": "file:../../../code/addons/interactions",
    "@storybook/addon-jest": "file:../../../code/addons/jest",
    "@storybook/addon-links": "file:../../../code/addons/links",
    "@storybook/addon-measure": "file:../../../code/addons/measure",
    "@storybook/addon-mgx-gfm": "file:../../../code/addons/mgx-gfm",
    "@storybook/addon-outline": "file:../../../code/addons/outline",
    "@storybook/addon-storysource": "file:../../../code/addons/storysource",
    "@storybook/addon-themes": "file:../../../code/addons/themes",
    "@storybook/addon-toolbars": "file:../../../code/addons/toolbars",
    "@storybook/addon-viewport": "file:../../../code/addons/viewport",
    "@storybook/angular": "file:../../../code/frameworks/angular",
    "@storybook/blocks": "file:../../../code/lib/blocks",
    "@storybook/builder-vite": "file:../../../code/builders/builder-vite",
    "@storybook/builder-webpack5": "file:../../../code/builders/builder-webpack5",
    "@storybook/codemod": "file:../../../code/lib/codemod",
    "storybook": "portal:../../../code/core",
    "@storybook/core-webpack": "portal:../../../code/lib/core-webpack",
    "@storybook/csf-plugin": "portal:../../../code/lib/csf-plugin",
    "@storybook/ember": "portal:../../../code/frameworks/ember",
    "@storybook/addon-test": "file:../../../code/addons/test",
    "@storybook/html": "file:../../../code/renderers/html",
    "@storybook/html-webpack5": "file:../../../code/frameworks/html-webpack5",
    "@storybook/instrumenter": "file:../../../code/lib/instrumenter",
    "@storybook/nextjs": "file:../../../code/frameworks/nextjs",
    "@storybook/preact": "file:../../../code/renderers/preact",
    "@storybook/preact-vite": "file:../../../code/frameworks/preact-vite",
    "@storybook/preact-webpack5": "file:../../../code/frameworks/preact-webpack5",
    "@storybook/preset-create-react-app": "file:../../../code/presets/create-react-app",
    "@storybook/preset-html-webpack": "file:../../../code/presets/html-webpack",
    "@storybook/preset-preact-webpack": "file:../../../code/presets/preact-webpack",
    "@storybook/preset-react-webpack": "file:../../../code/presets/react-webpack",
    "@storybook/preset-server-webpack": "file:../../../code/presets/server-webpack",
    "@storybook/preset-svelte-webpack": "file:../../../code/presets/svelte-webpack",
    "@storybook/preset-vue3-webpack": "file:../../../code/presets/vue3-webpack",
    "@storybook/react": "file:../../../code/renderers/react",
    "@storybook/react-dom-shim": "file:../../../code/lib/react-dom-shim",
    "@storybook/react-vite": "file:../../../code/frameworks/react-vite",
    "@storybook/react-webpack5": "file:../../../code/frameworks/react-webpack5",
    "@storybook/server": "file:../../../code/renderers/server",
    "@storybook/server-webpack5": "file:../../../code/frameworks/server-webpack5",
    "@storybook/source-loader": "file:../../../code/lib/source-loader",
    "@storybook/svelte": "file:../../../code/renderers/svelte",
    "@storybook/svelte-vite": "file:../../../code/frameworks/svelte-vite",
    "@storybook/svelte-webpack5": "file:../../../code/frameworks/svelte-webpack5",
    "@storybook/sveltekit": "file:../../../code/frameworks/sveltekit",
    "@storybook/test": "file:../../../code/lib/test",
    "@storybook/vue3": "file:../../../code/renderers/vue3",
    "@storybook/vue3-vite": "file:../../../code/frameworks/vue3-vite",
    "@storybook/vue3-webpack5": "file:../../../code/frameworks/vue3-webpack5",
    "@storybook/web-components": "file:../../../code/renderers/web-components",
    "@storybook/web-components-vite": "file:../../../code/frameworks/web-components-vite",
    "@storybook/web-components-webpack5": "file:../../../code/frameworks/web-components-webpack5",
    "playwright": "1.48.1"
  },
  "dependencies": {
    "react": "^18.0.0",
    "react-dom": "^18.0.0"
  },
  "devDependencies": {
    "@playwright/experimental-ct-react": "1.48.1",
    "@playwright/test": "1.48.1",
    "@storybook/addon-a11y": "^8.0.0",
    "@storybook/addon-actions": "^8.0.0",
    "@storybook/addon-controls": "^8.0.0",
    "@storybook/addon-essentials": "^8.0.0",
    "@storybook/addon-interactions": "^8.0.0",
    "@storybook/addon-test": "^8.0.0",
    "@storybook/react": "^8.0.0",
    "@storybook/react-vite": "^8.0.0",
    "@storybook/test": "^8.0.0",
    "@swc/core": "^1.4.2",
    "@swc/jest": "^0.2.36",
    "@testing-library/jest-dom": "6.6.3",
    "@testing-library/react": "^16.2.0",
    "@types/identity-obj-proxy": "^3",
    "@types/react": "^19.0.8",
    "@types/react-dom": "^19.0.3",
    "@typescript-eslint/eslint-plugin": "^6.21.0",
    "@typescript-eslint/parser": "^6.21.0",
    "@vitejs/plugin-react": "^4.2.1",
    "@vitest/browser": "^3.0.2",
    "@vitest/coverage-v8": "^3.0.2",
    "@vitest/ui": "^3.0.2",
    "cypress": "^13.6.4",
    "eslint": "^8.56.0",
    "eslint-plugin-react-hooks": "^4.6.0",
    "eslint-plugin-react-refresh": "^0.4.5",
    "eslint-plugin-storybook": "^0.6.15",
    "identity-obj-proxy": "^3.0.0",
    "jest": "^29.7.0",
    "jest-environment-jsdom": "^29.7.0",
    "storybook": "^8.0.0",
    "typescript": "^5.7.3",
    "vite": "^5.1.1",
    "vitest": "^3.0.2"
  }
}