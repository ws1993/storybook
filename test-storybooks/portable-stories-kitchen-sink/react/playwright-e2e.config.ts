import { defineConfig, devices } from '@playwright/test';
import path from 'node:path';

/**
 * See https://playwright.dev/docs/test-configuration.
 */
export default defineConfig({
  testDir: './e2e-tests',
  outputDir: './test-results',
  /* Maximum time one test can run for. */
  timeout: (process.env.CI ? 60 : 30) * 1000,
  /* Run tests in files in parallel */
  fullyParallel: false,
  /* Fail the build on CI if you accidentally left test.only in the source code. */
  forbidOnly: !!process.env.CI,
  retries: 0,
  /* Run tests serially to avoid side effects */
  workers: 1,
  /* Reporter to use. See https://playwright.dev/docs/test-reporters */
  reporter: [
    ['line'],
    [
      'junit',
      {
        embedAnnotationsAsProperties: true,
        outputFile: path.join(__dirname, '..','..','..', 'test-results'),
      },
    ],
  ],
  /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
  use: {
    /* Collect trace when retrying the failed test. See https://playwright.dev/docs/trace-viewer */
    trace: 'retain-on-failure',
    video: 'retain-on-failure',
    // headless: false,
  },

  /* Configure projects for major browsers */
  projects: [
    {
      name: 'chromium',
      use: { ...devices['Desktop Chrome'] },
    },
    // {
    //   name: 'firefox',
    //   use: { ...devices['Desktop Firefox'] },
    // },
    // {
    //   name: 'webkit',
    //   use: { ...devices['Desktop Safari'] },
    // },
  ],

  webServer: {
    command: 'yarn storybook',
    url: 'http://127.0.0.1:6006',
    reuseExistingServer: true,
    stdout: 'pipe',
    stderr: 'pipe',
  },
});
