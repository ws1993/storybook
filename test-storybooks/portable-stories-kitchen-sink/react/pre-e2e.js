/* eslint-disable no-undef */
/* eslint-disable @typescript-eslint/no-var-requires */
const fs = require("node:fs");
const path = require("node:path");

const testStoryPath = path.resolve("stories/AddonTest.stories.tsx");

console.log(`Pre-e2e script: clearing ${testStoryPath}`);
const storyContent = fs.readFileSync(testStoryPath).toString();
fs.writeFileSync(testStoryPath, storyContent.replace("forceFailure: true", "forceFailure: false"));