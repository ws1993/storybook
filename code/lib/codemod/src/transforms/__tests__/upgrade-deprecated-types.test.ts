import { describe, expect, it } from 'vitest';

import ansiRegex from 'ansi-regex';
import type { API } from 'jscodeshift';
import { dedent } from 'ts-dedent';

import _transform from '../upgrade-deprecated-types';

expect.addSnapshotSerializer({
  serialize: (val: any) => (typeof val === 'string' ? val : val.toString()),
  test: () => true,
});

const tsTransform = async (source: string) =>
  (
    await _transform({ source, path: 'Component.stories.tsx' }, {} as API, {
      parser: 'tsx',
    })
  ).trim();

describe('upgrade-deprecated-types', () => {
  describe('typescript', () => {
    it('upgrade regular imports', async () => {
      await expect(
        tsTransform(dedent`
          import { Story, ComponentMeta, Meta, ComponentStory, ComponentStoryObj, ComponentStoryFn } from '@storybook/react';
          import { Cat, CatProps } from './Cat';
            
          const meta = { title: 'Cat', component: Cat } satisfies ComponentMeta<typeof Cat>
          const meta2: Meta<CatProps> = { title: 'Cat', component: Cat };
          export default meta;
          
          export const A: ComponentStory<typeof Cat> = (args) => <Cat {...args} />;
          export const B: any = (args) => <Button {...args} />;
          export const C: ComponentStoryFn<typeof Cat> = (args) => <Cat {...args} />;
          export const D: ComponentStoryObj<typeof Cat> = {
            args: {
              name: 'Fluffy',
            },
          };
          export const E: Story<CatProps> = (args) => <Cat {...args} />;
        `)
      ).resolves.toMatchSnapshot();
    });

    it('upgrade imports with local names', async () => {
      await expect(
        tsTransform(dedent`
          import { Story as Story_, ComponentMeta as ComponentMeta_, ComponentStory as Story__, ComponentStoryObj as ComponentStoryObj_, ComponentStoryFn as StoryFn_ } from '@storybook/react';
          import { Cat } from './Cat';
            
          const meta = { title: 'Cat', component: Cat } satisfies ComponentMeta_<typeof Cat>
          const meta2: ComponentMeta_<typeof Cat> = { title: 'Cat', component: Cat };
          export default meta;
          
          export const A: Story__<typeof Cat> = (args) => <Cat {...args} />;
          export const B: any = (args) => <Button {...args} />;
          export const C: StoryFn_<typeof Cat> = (args) => <Cat {...args} />;
          export const D: ComponentStoryObj_<typeof Cat> = {
            args: {
              name: 'Fluffy',
            },
          };
          export const E: Story_<CatProps> = (args) => <Cat {...args} />;
        `)
      ).resolves.toMatchSnapshot();
    });

    it('upgrade imports with conflicting local names', async () => {
      await expect.addSnapshotSerializer({
        serialize: (value) => {
          const stringVal = typeof value === 'string' ? value : value.toString();
          return stringVal.replace(ansiRegex(), '');
        },
        test: () => true,
      });

      await expect(() =>
        tsTransform(dedent`
          import { ComponentMeta as Meta, ComponentStory as StoryFn } from '@storybook/react';
          import { Cat } from './Cat';
            
          const meta = { title: 'Cat', component: Cat } satisfies Meta<typeof Cat>
          export default meta;
          
          export const A: StoryFn<typeof Cat> = (args) => <Cat {...args} />;
         
        `)
      ).rejects.toThrowErrorMatchingSnapshot();
    });

    it('upgrade namespaces', async () => {
      await expect(
        tsTransform(dedent`
          import * as SB from '@storybook/react';
          import { Cat, CatProps } from './Cat';
  
          const meta = { title: 'Cat', component: Cat } satisfies SB.ComponentMeta<typeof Cat>;
          const meta2: SB.ComponentMeta<typeof Cat> = { title: 'Cat', component: Cat };
          export default meta;
  
          export const A: SB.ComponentStory<typeof Cat> = (args) => <Cat {...args} />;
          export const B: any = (args) => <Button {...args} />;
          export const C: SB.ComponentStoryFn<typeof Cat> = (args) => <Cat {...args} />;
          export const D: SB.ComponentStoryObj<typeof Cat> = {
            args: {
              name: 'Fluffy',
            },
          };
          export const E: SB.Story<CatProps> = (args) => <Cat {...args} />;
          
        `)
      ).resolves.toMatchSnapshot();
    });
  });
});
