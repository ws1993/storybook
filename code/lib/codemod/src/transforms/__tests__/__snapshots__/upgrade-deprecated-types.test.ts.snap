// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`upgrade-deprecated-types > typescript > upgrade imports with conflicting local names 1`] = `
SyntaxError: This codemod does not support local imports that are called the same as a storybook import.
Rename this local import and try again.
> 1 |  import { ComponentMeta as Meta, ComponentStory as StoryFn } from '@storybook/react';
    |           ^^^^^^^^^^^^^^^^^^^^^
  2 |  import { Cat } from './Cat';
  3 |    
  4 |  const meta = { title: 'Cat', component: Cat } satisfies Meta<typeof Cat>
`;

exports[`upgrade-deprecated-types > typescript > upgrade imports with local names 1`] = `
import {
  Meta as ComponentMeta_,
  StoryObj as ComponentStoryObj_,
  StoryFn as Story_,
} from '@storybook/react';

import { Cat } from './Cat';

const meta = { title: 'Cat', component: Cat } satisfies ComponentMeta_<typeof Cat>;
const meta2: ComponentMeta_<typeof Cat> = { title: 'Cat', component: Cat };
export default meta;

export const A: Story__<typeof Cat> = (args) => <Cat {...args} />;
export const B: any = (args) => <Button {...args} />;
export const C: StoryFn_<typeof Cat> = (args) => <Cat {...args} />;
export const D: ComponentStoryObj_<typeof Cat> = {
  args: {
    name: 'Fluffy',
  },
};
export const E: Story_<CatProps> = (args) => <Cat {...args} />;
`;

exports[`upgrade-deprecated-types > typescript > upgrade namespaces 1`] = `
import * as SB from '@storybook/react';

import { Cat, CatProps } from './Cat';

const meta = { title: 'Cat', component: Cat } satisfies SB.Meta<typeof Cat>;
const meta2: SB.Meta<typeof Cat> = { title: 'Cat', component: Cat };
export default meta;

export const A: SB.StoryFn<typeof Cat> = (args) => <Cat {...args} />;
export const B: any = (args) => <Button {...args} />;
export const C: SB.StoryFn<typeof Cat> = (args) => <Cat {...args} />;
export const D: SB.StoryObj<typeof Cat> = {
  args: {
    name: 'Fluffy',
  },
};
export const E: SB.StoryFn<CatProps> = (args) => <Cat {...args} />;
`;

exports[`upgrade-deprecated-types > typescript > upgrade regular imports 1`] = `
import { Meta, StoryFn, StoryObj } from '@storybook/react';

import { Cat, CatProps } from './Cat';

const meta = { title: 'Cat', component: Cat } satisfies Meta<typeof Cat>;
const meta2: Meta<CatProps> = { title: 'Cat', component: Cat };
export default meta;

export const A: StoryFn<typeof Cat> = (args) => <Cat {...args} />;
export const B: any = (args) => <Button {...args} />;
export const C: StoryFn<typeof Cat> = (args) => <Cat {...args} />;
export const D: StoryObj<typeof Cat> = {
  args: {
    name: 'Fluffy',
  },
};
export const E: StoryFn<CatProps> = (args) => <Cat {...args} />;
`;
